app/models/user.rb

```diff
@@ -179,12 +178,41 @@ def resources
     user_permissions.map(&:resource)
   end
 
+  # #########################
+  # User Access (permissions)
+  #
+  def accessible_organizations(action)
+    return Organization.all if power_user?
+    accesses.organizations(action)
+  end
+
+  def accessible_facility_groups(action)
+    return FacilityGroup.all if power_user?
+    accesses.facility_groups(action)
+  end
+
+  def accessible_facilities(action)
+    return Facility.all if power_user?
+    accesses.facilities(action)
+  end
+
+  def can?(action, model, record = nil)
+    return true if power_user?
+    accesses.can?(action, model, record)
+  end
+  #
+  # #########################
+
   def destroy_email_authentications
-    destroyable_email_authentications = email_authentications.load
+    destroyable_email_auths = email_authentications.load
```

Comment:
Haha was it really that bad? 😄

Reply:
RubyMine was constantly spitting warnings for E_TOO_LONG_VARIABLE_NAMES.